File: 01-Sorting/01-BubbleSort/01-BubbleSort.c
#include<stdio.h>
#include<time.h> 

#define ARRAY_LEN(x) (sizeof(x) / sizeof((x)[0]))

#define SWAP(a, b) ((&(a) == &(b)) ? (a) : ((a)^=(b),(b)^=(a),(a)^=(b)))

void BubbleSort(int arr[], int n) 
{
   int i, j; 
   for (i = 0; i < n-1; i++)
   {
      for (j = 0; j < n-i-1; j++)
      {
         if (arr[j] > arr[j+1])
            SWAP(arr[j], arr[j+1]);
      }  
   }
}

void PrintArray(int arr[], int n)
{
   int i;
   for (i=0; i < n; i++)
      printf("%d ", arr[i]);
   printf("\n");
}

int main()
{
   clock_t start_time, end_time;
   start_time = clock();

   int num_arr[] = {58, 24, 84, 2, 30, 27, 35, 91, 14, 23, 53, 33, 99, 17, 86};
   int len;

   len = ARRAY_LEN(num_arr);
   printf("Random Array: ");
   PrintArray(num_arr, len);
   printf("Performing Bubble Sort on Given Array.\n");
   BubbleSort(num_arr, len);
   printf("Sorted Array: ");
   PrintArray(num_arr, len);

   end_time = clock();

   double execution_time = (double)(end_time - start_time)* 1000.0 / CLOCKS_PER_SEC;
   printf("Code Execution Time: %f Milliseconds\n",  execution_time);

   return 0;
}

File: 01-Sorting/01-BubbleSort/01-BubbleSort.py
import time

def BubbleSort(arr, n):
   # Traverse through all array elements 
   for i in range(n):
      # Last i elements are already in place 
      for j in range(0, n-i-1):
         # traverse the array from 0 to n-i-1 
         # Swap if the element found is greater 
         # than the next element 
         if arr[j] > arr[j+1]:
            arr[j], arr[j+1] = arr[j+1], arr[j] 

start_time = time.time()

num_arr = [58, 24, 84, 2, 30, 27, 35, 91, 14, 23, 53, 33, 99, 17, 86]
l = len(num_arr)
print("Random Array:", *num_arr)
print('Performing Bubble Sort on Given Array.')
BubbleSort(num_arr, l)
print("Sorted Array:", *num_arr)

end_time = time.time()
execution_time = round(((end_time - start_time)*1000),6)
print("Code Execution Time: %s Milliseconds" % execution_time)